File: wiki_java/src/main/java/com/github/klee0kai/wiki/provide/identifiers/AppComponent.java
package com.github.klee0kai.wiki.provide.identifiers;

import com.github.klee0kai.stone.annotations.component.Component;

@Component(identifiers = {ScreenId.class, LoginId.class})
public interface AppComponent {

    SevenPlanetModule planetsModule();

    ThreadsModule threadsModule();

    PresentersModule presentersModule();

    FeaturePresenter featurePresenter(LoginId loginId, ScreenId screenId);

    void inject(FeatureScreen screen, LoginId loginId, ScreenId screenId);

}

File: wiki_kotlin/src/main/kotlin/com/github/klee0kai/wiki/provide/identifiers/AppComponent.kt
package com.github.klee0kai.wiki.provide.identifiers

import com.github.klee0kai.stone.annotations.component.Component

@Component(identifiers = [ScreenId::class, LoginId::class])
interface AppComponent {
    fun planetsModule(): SevenPlanetModule

    fun threadsModule(): ThreadsModule

    fun presentersModule(): PresentersModule

    fun featurePresenter(loginId: LoginId, screenId: ScreenId): FeaturePresenter

    fun inject(screen: FeatureScreen, loginId: LoginId, screenId: ScreenId)
}
